dynacred/src/credentialStructBS.ts

```diff
@@ -0,0 +1,508 @@
+            if (c.value !== undefined) {
+                const value = c.value instanceof Buffer ? c.value : Buffer.from(c.value);
+                orig.setAttribute(c.cred, c.attr, value);
+            } else {
+                orig.deleteAttribute(c.cred, c.attr);
+            }
+        }
+        this.setCredentialStruct(tx, orig);
+    }
+
+    setCredential(tx: CredentialTransaction, cred: Credential) {
+        const credStruct = this.getValue();
+        credStruct.setCredential(cred.name, cred);
+        this.setCredentialStruct(tx, credStruct);
+    }
+
+    setCredentialStruct(tx: CredentialTransaction, credStruct: CredentialStruct) {
+        const versionBuf = Buffer.alloc(4);
+        versionBuf.writeInt32LE(this.credPublic.version.getValue() + 1, 0);
+        credStruct.setAttribute(ECredentials.pub, EAttributesPublic.version, versionBuf);
+        tx.invoke(this.id, CredentialsInstance.contractID,
+            CredentialsInstance.commandUpdate, [
+                new Argument({
+                    name: CredentialsInstance.argumentCredential,
+                    value: credStruct.toBytes(),
+                }),
+            ],
+        );
+    }
+
+    getCredentialInstanceMapBS(name: ECredentials): CredentialInstanceMapBS {
+        return CredentialInstanceMapBS.fromScratch(this.getCredentialBS(name));
+    }
+}
+
+export class CredentialBS extends BehaviorSubject<Credential> {
+
+    static fromScratch(csbs: CredentialStructBS, name: ECredentials): CredentialBS {
+        return new CredentialBS(csbs, name,
+            ConvertBS(csbs, (cs) => cs.getCredential(name) || new Credential({name})));
+    }
+
+    private static getAttribute(cred: Credential, name: string): Buffer | undefined {
+        const attr = cred.attributes.find((a) => a.name.toString() === name);
+        if (attr === undefined) {
+            return undefined;
+        }
+        return attr.value;
+    }
+    constructor(private csbs: CredentialStructBS,
+                private cred: ECredentials, cbs: BehaviorSubject<Credential>) {
+        super(cbs.getValue());
+        cbs.subscribe(this);
+    }
+
+    getAttributeBufferBS(name: string): AttributeBufferBS {
+        const bs = this.getAttributeBS(name);
+        return new AttributeBufferBS(this, bs, name);
+    }
+
+    getAttributeStringBS(name: string): AttributeStringBS {
+        const bs = ConvertBS(this.getAttributeBS(name), (buf) => buf.toString());
+        return new AttributeStringBS(this, bs, name);
+    }
+
+    getAttributeLongBS(name: string): AttributeLongBS {
+        const bs = ConvertBS(this.getAttributeBS(name), (buf) => Long.fromBytesLE(Array.from(buf)));
+        return new AttributeLongBS(this, bs, name);
+    }
+
+    getAttributePointBS(name: string): AttributePointBS {
+        const bs = ConvertBS(this.getAttributeBS(name), (buf) => {
+            try {
+                return PointFactory.fromProto(buf);
+            } catch (e) {
+                return undefined;
```

Comment:
No, because the `getAttributeBS` returns a `Buffer.alloc(0)` if the attribute is not found. Which is the 'empty' attribute - except for the `Point`, where there is no such thing, as we don't even know which group we should use.

Changing `AttributePointBS` to return `Point|undefined`.